src/Symfony/Component/Config/Definition/Builder/NodeDefinition.php

```diff
@@ -127,7 +127,9 @@ public function getNode($forceRootNode = false)
         }
 
         $node = $this->createNode();
-        $node->setAttributes($this->attributes);
+        if ($node instanceof BaseNode) {
+            $node->setAttributes($this->attributes);
+        }
```

Comment:
`$this->createNode()` returns a `NodeInterface`, that does not define `setAttributes()`.

Reply:
does the current code prove this never happens in reallife? :sweat_smile:

Reply:
The current code in 4.4 assume that object implementing `NodeInterface` will extend `BaseNode`. That is wrong and this is what I show with the test. 

But yes, In the rare case that you create a new implementation of `NodeInterface`, you are using `BaseNode` 9/10.

Reply:
understood. but im not sure about silently ignoring $this->attributes etc. vs throwing for unepexted node types (not a BaseNode). As the crash would happen today as well.

Reply:
or fix it at the contract level ` @return BaseNode` on `createNode` perhsaps.

Reply:
If the `$node` does not have the concept of "attributes", then it is fine to ignore. 

The current code throws an exception like "Cannot call `setAttributes()` on class `CustomNode`".

Changing the contract on the abstract `NodeDefinition::createNode()` is a BC break. 

I like your suggestions but I do believe the current PR is what makes most sense.

Reply:
it figured, i agree with you :) IICU the 'consistent' way would be to move the setAttributes call to the appropriate child classes (where initNode gives a BaseNode), which is also a BC break likely.

:+1: